src/NServiceBus.AcceptanceTests/EndpointTemplates/PersistencelessServer.cs

```diff
@@ -0,0 +1,145 @@
+﻿namespace NServiceBus.AcceptanceTests.EndpointTemplates
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting.Customization;
+    using NServiceBus.AcceptanceTesting.Support;
+    using NServiceBus.Config.ConfigurationSource;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using Extensibility;
+    using NServiceBus.Features;
+    using NServiceBus.Hosting.Helpers;
+    using NServiceBus.ObjectBuilder;
+    using NServiceBus.Serialization;
+    using Unicast.Subscriptions;
+    using Unicast.Subscriptions.MessageDrivenSubscriptions;
+
+    public class PersistencelessServer : IEndpointSetupTemplate
+    {
+        public PersistencelessServer()
+        {
+            typesToInclude = new List<Type>();
+        }
+
+        public PersistencelessServer(List<Type> typesToInclude)
+        {
+            this.typesToInclude = typesToInclude;
+        }
+
+        public async Task<EndpointConfiguration> GetConfiguration(RunDescriptor runDescriptor, EndpointCustomizationConfiguration endpointConfiguration, IConfigurationSource configSource, Action<EndpointConfiguration> configurationBuilderCustomization)
+        {
+            var settings = runDescriptor.Settings;
+
+            var types = GetTypesScopedByTestClass(endpointConfiguration);
+
+            typesToInclude.AddRange(types);
+
+            var builder = new EndpointConfiguration();
+
+            builder.EndpointName(endpointConfiguration.EndpointName);
+            builder.TypesToIncludeInScan(typesToInclude);
+            builder.CustomConfigurationSource(configSource);
+            builder.EnableInstallers();
+
+            builder.DisableFeature<TimeoutManager>();
+            builder.DisableFeature<SecondLevelRetries>();
+            builder.DisableFeature<FirstLevelRetries>();
+
+            await builder.DefineTransport(settings, endpointConfiguration.EndpointName).ConfigureAwait(false);
+
+            builder.DefineBuilder(settings);
+            builder.RegisterComponents(r =>
+            {
+                RegisterInheritanceHierarchyOfContextOnContainer(runDescriptor, r);
+                r.ConfigureComponent<ISubscriptionStorage>(() => new NoOpISubscriptionStorage(), DependencyLifecycle.SingleInstance);
```

Comment:
If its persitence less why do we register a persitence? (could we move this the actual test instead perhaps?)